### PR TITLE
Add Promotion to Production step to Drone

### PR DESCRIPTION
**Pull request by kaxil**
How it works?

Once the build passes on the master branch and a git tag is released, a `terraform_plan_staging_deployment` will be run in drone CI and it would show the the resources that would be created.

To approve once you confirm the plan we will have to run :

```
drone build promote astronomer/terraform-google-astronomer-cloud ${DRONE_BUILD_NUMBER} staging
```

This will run the `deploy_to_staging` & `slack_staging` tasks.

### CODE REVIEW

File: pipeline/release_stage_cloud.sh [hunk 1]
```diff
@@ -27,27 +27,37 @@ sed -i "s/PROJECT/astronomer-cloud-staging/g" providers.tf
...
-  --target=module.astronomer_cloud.local_file.kubeconfig
-
-terraform apply --auto-approve \
-  -var "deployment_id=$DEPLOYMENT_ID" \
-  -var "dns_managed_zone=staging-zone" \
-  -var "zonal=$ZONAL" \
-  -lock=false \
-  -refresh=false
+if [[ ${TF_PLAN:-0} -eq 1 ]]; then
+	terraform plan -detailed-exitcode \
```
Comment (sjmiller609) [inline pipeline/release_stage_cloud.sh]: -out

File: pipeline/release_stage_cloud.sh [hunk 2]
```diff
@@ -27,27 +27,37 @@ sed -i "s/PROJECT/astronomer-cloud-staging/g" providers.tf
...
+fi
+
+if [[ ${TF_APPLY:-0} -eq 1 ]]; then
+	terraform apply --auto-approve \
+	  -var "deployment_id=$DEPLOYMENT_ID" \
+	  -var "dns_managed_zone=staging-zone" \
+	  -var "zonal=$ZONAL" \
+	  -lock=false \
+	  --target=module.astronomer_cloud.module.gcp
+
```
Comment (sjmiller609) [inline pipeline/release_stage_cloud.sh]: add 'plan' target to make sure that plan correct

File: pipeline/release_stage_cloud.sh (hunk 1 above)
Comment (kaxil) [reply pipeline/release_stage_cloud.sh]: Why do we need to save the plan to the file?

Comment (kaxil) [reply pipeline/release_stage_cloud.sh]: updated

File: pipeline/release_stage_cloud.sh (hunk 2 above)
Comment (kaxil) [reply pipeline/release_stage_cloud.sh]: done

**Pull request merged**